modules/primo/src/Primo/BriefSearch/Client.php

```diff
@@ -134,13 +134,20 @@ public function search($query, $offset, $numResults, $parameters = []) {
    */
   public function validateThumbnail($url) {
     // Use HEAD as we only care about status code and content type here.
-    $response = $this->httpClient->head($url);
-    // According to docs content type should be a string but is really a array
-    // with a single string entry.
-    /* @var string[] $contentType */
-    $contentType = $response->getHeader('Content-Type');
-    return $response->getStatusCode() === 200 &&
-      stripos(implode('', $contentType), 'image') === 0;
+    try {
+      $response = $this->httpClient->head($url);
+      // According to docs content type should be a string but is really a array
+      // with a single string entry.
+      /* @var string[] $contentType */
+      $contentType = $response->getHeader('Content-Type');
+      return stripos(implode('', $contentType), 'image') === 0;
+    }
+    catch (RequestException $e) {
+      if ($e->getResponse()->getStatusCode() != 404) {
```

Comment:
Why only throw the exception if the status code is 404?
I'd expect `!== 200`?

Reply:
As I understand it exceptions are thrown for non-succesfull responses so status code will always be different from 200 here.

The reason we check for 404 is that 404 will signal a nonexistant cover. That situation is OK and we jusgt return `false`. If other types of exceptions occur we want to log them as well.